python/src/random.cpp

```diff
@@ -309,4 +358,7 @@ void init_random(py::module_& parent_module) {
         Returns:
             array: The ``shape``-sized output array with type ``uint32``.
       )pbdoc");
+  // Register static Python object cleanup before the interpreter exits
+  auto atexit = py::module_::import("atexit");
+  atexit.attr("register")(py::cpp_function([]() { default_key().release(); }));
```

Comment:
release and dec ref perhaps?

Reply:
Oops!

Reply:
Actually, `release()` is a function I wrote that does the dec ref on the underlying Python object 😅